src/Autofac.Extras.Moq/MoqRegistrationHandler.cs

```diff
@@ -126,6 +127,7 @@ public MoqRegistrationHandler(ISet<Type> createdServiceTypes, ISet<Type> mockedS
                     // Their constructor dependencies will then be mocked.
                     result = RegistrationBuilder.ForType(typedService.ServiceType)
                                             .InstancePerLifetimeScope()
+                                            .ExternallyOwned()
```

Comment:
This one should not be ExternallyOwned, because it is the actual service under test (and may have legitimate dispose behaviour), rather than the mock.

Reply:
I fixed this, along with a test that should cover this case.